### PR TITLE
add reshape frontends.torch

### PR DESCRIPTION
**Pull request by PatternFinder**
link to issue: #2636

reason for new PR: cleaned up local branch name formatting

### CODE REVIEW

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @PatternFinder, thanks for contributing. There are a few changes to be made to your PR before we get it merged. I have suggested changes near the lines of code concerned. Please feel free to re-request my review once you're done making the changes, we'll look into getting the tests passing after that.
Let me know if you have any questions. Thanks 🙂

**Comment by PatternFinder** [comment]
> 

@VedPatwardhan got it, did all the things per your instructions. ready for review :)

File: ivy/functional/frontends/tensorflow/functions.py [hunk 1]
```diff
@@ -27,16 +27,22 @@ def fill(dims, value, name="full"):
 fill.unsupported_dtypes = {"torch": ("float16", "bfloat16")}
 
 
-
```
Comment (VedPatwardhan) [inline ivy/functional/frontends/tensorflow/functions.py:30]: Please revert all changes made to this file as this isn't related to the function in this PR.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tf_functions.py [hunk 2]
```diff
@@ -193,10 +193,10 @@ def test_tensorflow_full(
 
 # mutiply
 @given(
-        dtype_and_x=helpers.dtype_and_values(
```
[the remark this answers is not in the document]
Comment (VedPatwardhan) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tf_functions.py:196]: These changes haven't been reverted yet, please revert them.

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py [hunk 3]
```diff
@@ -4,3 +4,14 @@
 
 def cat(tensors, dim=0, *, out=None):
     return ivy.concat(tensors, dim, out=out)
+
+
+def reshape(input, shape, copy=None):
```
Comment (VedPatwardhan) [inline ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:9]: The `copy` argument isn't accepted by [`torch.reshape`](https://pytorch.org/docs/stable/generated/torch.reshape.html), hence we shouldn't have it here.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 4]
```diff
@@ -70,3 +71,43 @@ def test_torch_cat(
...
+@given(
+    dtype_value_shape=helpers.dtype_and_values(
+        available_dtypes=tuple(
+            set(ivy_np.valid_float_dtypes).intersection(
+                set(ivy_torch.valid_float_dtypes)
+            )
+        ),
+        ret_shape=True,
+    ),
+    copy=st.booleans(),
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:86]: Please remove the usage of the `copy` argument from the `test_torch_reshape` function too.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 5]
```diff
@@ -70,3 +71,43 @@ def test_torch_cat(
...
+def test_torch_reshape(
+    dtype_value_shape,
+    copy,
+    as_variable,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    input_dtype, value, shape = dtype_value_shape
+    helpers.test_frontend_function(
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:102]: Please also pass `with_out=False` to the `test_frontend_function` here.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @PatternFinder, thanks for making the changes, I have suggested a few more changes and also mentioned the ones from the previous review that weren't made, please make those changes and re-request my review once you're done.
Let me know if you have any questions. Thanks 🙂

**Comment by PatternFinder** [comment]
ok so here's what happend :)
1. the tf codes that were not part of my task I originally edited them to attempt to fix all the linting errors
2. but now I'm so far behind upstream so I decided just discard all my changes and remerge with upstream (which closed this PR :(
3. I've edited the reshape function per your instructions and added it back to this branch, and opened another PR for the reshape function at #2853

thanks for your support!

**Pull request closed without merging**